mypy/test/testcheck.py

```diff
@@ -149,15 +149,15 @@ def run_case_once(self, testcase: DataDrivenTestCase, incremental=0) -> None:
         sources = []
         for module_name, program_path, program_text in module_data:
             # Always set to none so we're forced to reread the module in incremental mode
-            program_text = None if incremental else program_text
-            sources.append(BuildSource(program_path, module_name, program_text))
+            sources.append(BuildSource(program_path, module_name,
+                                       None if incremental else program_text))
+        res = None
```

Comment:
This refactor was needed because program_text is previously inferred as str, and that's correct. So this is technically a case of reusing the same variable name with a different type (#1174).